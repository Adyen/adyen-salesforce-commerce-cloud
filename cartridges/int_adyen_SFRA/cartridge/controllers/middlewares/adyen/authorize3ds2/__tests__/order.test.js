"use strict";

jest.mock('../errorHandler', function () {
  return {
    handlePlaceOrderError: jest.fn()
  };
});
var handlePlaceOrder;
var COHelpers;
var req;
var res;
var adyenHelper;
beforeEach(function () {
  jest.clearAllMocks();
  req = {
    locale: {
      id: 'mocked_locale_id'
    }
  };
  res = {
    redirect: jest.fn(),
    render: jest.fn()
  };
  handlePlaceOrder = require("../order");
  COHelpers = require('*/cartridge/scripts/checkout/checkoutHelpers');
  adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
});
describe('Order', function () {
  it('should handle place order error', function () {
    var _require = require('../errorHandler'),
        handlePlaceOrderError = _require.handlePlaceOrderError;

    adyenHelper.getAdyenSFRA6Compatibility.mockReturnValue(true);
    COHelpers.placeOrder.mockReturnValue({
      error: true
    });
    handlePlaceOrder();
    expect(handlePlaceOrderError).toBeCalledTimes(1);
  });
  it('should handle place order error', function () {
    var OrderMgr = require('dw/order/OrderMgr');

    adyenHelper.getAdyenSFRA6Compatibility.mockReturnValue(true);
    var order = OrderMgr.getOrder(session.privacy.orderNo);
    var paymentInstrument = order.getPaymentInstruments()[0];
    handlePlaceOrder(paymentInstrument, order, {}, {
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(res.render.mock.calls).toMatchSnapshot();
  });
});