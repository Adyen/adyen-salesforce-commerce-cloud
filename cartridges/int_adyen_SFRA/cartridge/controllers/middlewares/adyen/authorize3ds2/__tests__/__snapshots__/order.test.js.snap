// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should handle place order error 1`] = `
Array [
  Array [
    "orderConfirmForm",
    Object {
      "orderID": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
    },
  ],
]
`;

exports[`Order should handle place order error 1`] = `Array []`;
